File: Coding-Interviews/28.对称的二叉树.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def isSymmetric(self, root: TreeNode) -> bool:
        def isSame(root1, root2):
            if not root2 and not root1:
                return True
            if not root2 or not root1 or root1.val != root2.val:
                return False
            return isSame(root1.left, root2.right) and isSame(root1.right, root2.left)

        return isSame(root.right, root.left) if root else True

File: Coding-Interviews/28.对称的二叉树.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
#include <ios>
class Solution
{
public:
    bool helper(TreeNode* left, TreeNode* right)
    {
        if (!left && !right)
            return true;
        if (!left || !right)
            return false;
        return left->val == right->val && helper(left->left, right->right) &&
               helper(left->right, right->left);
    }
    bool isSymmetric(TreeNode* root)
    {
        if (!root)
            return true;
        return helper(root->left, root->right);
    }
};
